### ISSUE TITLE
Restore iterative implementations of ord=1

### ISSUE THREAD

**Issue by goodfeli** [cleverhans-lab/cleverhans#830]
This PR ( https://github.com/tensorflow/cleverhans/pull/829 ) turned them iterative attacks that use ord=1 because they aren't known to be strong. To turn them back one:

- We should implement correct projection to the norm=1 ball, or come up with some other way to optimize over the norm=1 ball that we know is at least as effective as textbook PGD.
- We should check whether it's better to literally do gradient descent or whether it's better to do iterated FGM. If the local linear approximation is valid, FGM will raise the loss more than any other step with the same L1 norm, but it's not clear that that's the right way to think about the problem. For the case of ord=1, it seems weird to use an inner loop step that can change only one pixel at a time.

(Note that we do have an optimal implementation of the single step ord=1 attack)

**Comment by ftramer** [comment]
This PR (https://github.com/tensorflow/cleverhans/pull/1036) implements the correct projection onto the norm=1 ball using Duchi et al.'s O(d*log(d)) algorithm.
It also implements a new multi-step attack that is much more efficient that iterated FGM for the l1-case, simply by relaxing the sparsity of the gradient updates.

**Comment by jonasguan** [comment]
Closing as we update the library to v4 and drop support for version < 4.
